NCMP_for_2-1-1_and_2-2-1/WHO_code_toRUN.R

```diff
@@ -0,0 +1,96 @@
+###############################################################################################
+# 
+# The user must set the working directory to match the directory of this R script
+#
+# setwd() #! uncomment this line to set the directory of the downloded script.
+#
+# input: .csv file with NCMP data (separate file for each year) with the following variables present:
+# [GenderCode]	    Sex of child coded as M and F
+# [AgeInMonths]	    Age at time of measurement 
+# [Ethnicity_desc]	17 ethnic groups, text descriptions e.g.  White - British
+# [Height]	        Height in cm 
+# [Weight]	        Weight in kg 
+# [PupilIndexOfMultipleDeprivationD]	IMD decile of child LSOA
+# [PupilUrbanRuralIndicator]	ONS urban/rural classification based on child LSOA - code
+#
+# 
+# NB: The WHO function needs to be run on each year of data separately
+# The code below performs some pre-processing before feeding the data to the WHO function,
+# and post-processing before outputting a single csv with the final data for the corresponding year
+# The output file is located one directory up from the current working dir.
+###############################################################################################
+
+
+# install the package and load it in your R session (only need to do this once):
+# install.packages("anthro",dependencies=TRUE, type="win.binary") 
+library(anthro) #need to run every time you start a new session, if you want to use the package
+
+NCMP_data <- read.csv("../NCMP_data.csv") #change file name and location accordingly
+
+# rounding/truncating the age in months variable
+NCMP_data$AgeInMonths <- trunc(NCMP_data$AgeInMonths) 
+
+# retaining only age groups of interest (under 5s, exclusing 5)
+NCMP_data <- NCMP_data[NCMP_data$AgeInMonths<60, ]
+
+
+# regroup imd variable from deciles into quintiles (pair in twos)
+# overwrite 2s with 1, 4s and 3s with 2, 5s and 6s with 3, 7s and 8s with 4, and 9s and 10s with 5
+NCMP_data$PupilIndexOfMultipleDeprivationD <- ifelse(NCMP_data$PupilIndexOfMultipleDeprivationD==1, 1, 
```

Comment:
Just realised an even easier way to redo this in one line!! Divide the deciles by two and round with ceiling (got to test it, but should work)! 

`clean_NCMP$school_imd_quintile <- ceiling(clean_NCMP$schoolindexofmultipledepriv/2)`